BootloaderCommonPkg/Library/LiteFvLib/FvLib.c

```diff
@@ -87,6 +116,95 @@ GetFfsFileByName (
   return EFI_NOT_FOUND;
 }
 
+
+/**
+  Find a file within a volume by its type.
+
+  This service searches for files with a specific name, within
+  either the specified firmware volume or all firmware volumes.
+
+  @param[in]       FvHeader    Pointer to a firmware volume header
+  @param[in]       FileType    The file type to search for.
+  @param[in]       Instance    0 based instance of the matched file.
+  @param[in, out]  File        Return pointer.  In the case of an error,
+                               contents are undefined.
+
+  @retval EFI_SUCCESS             The function completed successfully.
+  @retval EFI_ABORTED             An error was encountered.
+  @retval EFI_NOT_FOUND           File not found.
+  @retval EFI_INVALID_PARAMETER   One of the parameters was NULL.
+
+**/
+EFI_STATUS
+GetFfsFileByType (
+  IN  EFI_FIRMWARE_VOLUME_HEADER      *FvHeader,
+  IN  EFI_FV_FILETYPE                  FileType,
+  IN  UINT32                           Instance,
+  IN  OUT EFI_FFS_FILE_HEADER        **File
+  )
+{
+  EFI_FFS_FILE_HEADER             *CurrentFile;
+  EFI_FFS_FILE_HEADER             *NextFile;
+  EFI_FFS_FILE_HEADER             *EndFile;
+  UINT32                           Count;
+  EFI_FIRMWARE_VOLUME_EXT_HEADER  *FvExHeader;
+
+  //
+  // Verify library has been initialized.
+  //
+  if (FvHeader == NULL || File == NULL) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  if (FvHeader->Signature != EFI_FVH_SIGNATURE) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  //
+  // Get the first file
+  //
+  CurrentFile = (EFI_FFS_FILE_HEADER *) ((UINTN)FvHeader + FvHeader->HeaderLength);
+  if (FvHeader->ExtHeaderOffset != 0) {
+    FvExHeader  = (EFI_FIRMWARE_VOLUME_EXT_HEADER *)(((UINT8 *)FvHeader) + FvHeader->ExtHeaderOffset);
+    CurrentFile = (EFI_FFS_FILE_HEADER *)(((UINT8 *)FvExHeader) + FvExHeader->ExtHeaderSize);
+  }
+
+  //
+  // Loop as long as we have a valid file
+  //
+  Count = 0;
+  EndFile = (EFI_FFS_FILE_HEADER *) (UINTN) ((UINTN)FvHeader + FvHeader->FvLength) - 1;
+  while (CurrentFile != NULL) {
+    if (CurrentFile->Type == FileType) {
+      if (Count == Instance) {
+        *File = CurrentFile;
+        return EFI_SUCCESS;
+      }
+      Count++;
+    }
+
+    //
+    // Get next file, compensate for 8 byte alignment if necessary.
+    //
+    NextFile = (EFI_FFS_FILE_HEADER *) ((((UINTN)CurrentFile - (UINTN)FvHeader + \
```

Comment:
Can we simplify this to something like `ALIGN_UP(CurrentFile + GET_FFS_LENGTH(CurrentFile), 8)`